api/src/tests/state_test.rs

```diff
@@ -85,10 +95,167 @@ async fn test_get_account_module_not_found() {
+        )
+        .await["data"]
+        .to_owned();
+
+    assert_table_item(ctx, &tt["u8_table"], "u8", "u8", 1u8, 1u8).await;
+    assert_table_item(ctx, &tt["u64_table"], "u64", "u64", "1", "1").await;
+    assert_table_item(ctx, &tt["u128_table"], "u128", "u128", "1", "1").await;
+    assert_table_item(ctx, &tt["bool_table"], "bool", "bool", true, true).await;
+    assert_table_item(
+        ctx,
+        &tt["address_table"],
+        "address",
+        "address",
+        "0x1",
+        "0x1",
+    )
+    .await;
+    assert_table_item(
+        ctx,
+        &tt["string_table"],
+        "0x1::ASCII::String",
+        "0x1::ASCII::String",
+        "abc",
+        "abc",
+    )
+    .await;
+    assert_table_item(
+        ctx,
+        &tt["vector_u8_table"],
+        "vector<u8>",
+        "vector<u8>",
+        "0x0102",
+        "0x0102",
+    )
+    .await;
+    assert_table_item(
+        ctx,
+        &tt["vector_string_table"],
+        "vector<0x1::ASCII::String>",
+        "vector<0x1::ASCII::String>",
+        ["abc", "abc"],
+        ["abc", "abc"],
+    )
+    .await;
+    let id = &tt["id_table_id"];
+    assert_table_item(
+        ctx,
+        &tt["id_table"],
+        "0x1::GUID::ID",
+        "0x1::GUID::ID",
+        id,
+        id,
+    )
+    .await;
+    let nested_table = api_get_table_item(
+        ctx,
+        &tt["table_table"],
+        "u8",
+        "0x1::Table::Table<u8, u8>",
+        1u8,
+    )
+    .await;
+    assert_table_item(ctx, &nested_table, "u8", "u8", 2, 3).await;
+}
+
 fn get_account_resource(address: &str, struct_tag: &str) -> String {
     format!("/accounts/{}/resource/{}", address, struct_tag)
 }
 
 fn get_account_module(address: &str, name: &str) -> String {
     format!("/accounts/{}/module/{}", address, name)
 }
+
+fn get_table_item(handle: u128) -> String {
+    format!("/tables/{}/item", handle)
+}
+
+async fn make_test_tables(ctx: &mut TestContext, account: &mut LocalAccount) {
+    let module = build_test_module(account.address()).await;
+
+    ctx.api_publish_module(account, module.try_into().unwrap())
+        .await;
+    ctx.api_execute_script_function(
+        account,
+        "TableTestData::make_test_tables",
+        json!([]),
+        json!([]),
+    )
+    .await
+}
+
+async fn build_test_module(account: AccountAddress) -> Vec<u8> {
+    let package_dir = PathBuf::from(std::env!("CARGO_MANIFEST_DIR"))
+        .parent()
+        .unwrap()
+        .join("api/move-test-package");
+    let build_config = BuildConfig {
+        generate_docs: false,
+        install_dir: Some(package_dir.clone()),
+        additional_named_addresses: [("TestAccount".to_string(), account)].into(),
+        ..Default::default()
+    };
+    let _package = build_config
+        .compile_package(&package_dir, &mut Vec::new())
+        .unwrap();
+
+    let mut out = Vec::new();
+    tokio::fs::File::open(package_dir.join("build/ApiTest/bytecode_modules/TableTestData.mv"))
```

Comment:
I don't think you need to read from file, it's in the compiled package. see https://github.com/aptos-labs/aptos-core/blob/b1601d23fb157fee4c4932cd093d3809be8094a4/testsuite/smoke-test/src/aptos/module_publish.rs#L34

Reply:
👌 (But we'll serialize it again!)